drivers/serial/uart_msp432p4xx.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (c) 2017, Linaro Ltd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* See www.ti.com/lit/pdf/slau356f, Chapter 22, for MSP432P4XX UART info. */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <uart.h>
+
+/* Driverlib includes */
+#include <driverlib/rom.h>
+#include <driverlib/rom_map.h>
+#include <driverlib/gpio.h>
+#include <driverlib/uart.h>
+#include <driverlib/interrupt.h>
+
+struct uart_msp432p4xx_dev_data_t {
+	/* UART config structure */
+	eUSCI_UART_Config uartConfig;
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	uart_irq_callback_t cb; /**< Callback function pointer */
+#endif /* CONFIG_UART_INTERRUPT_DRIVEN */
+};
+
+#define DEV_CFG(dev) \
+	((const struct uart_device_config * const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct uart_msp432p4xx_dev_data_t * const)(dev)->driver_data)
+
+static struct device DEVICE_NAME_GET(uart_msp432p4xx_0);
+
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static void uart_msp432p4xx_isr(void *arg);
+#endif
+
+static const struct uart_device_config uart_msp432p4xx_dev_cfg_0 = {
+	.base = (void *)EUSCI_A0_BASE,
+	.sys_clk_freq = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
+};
+
+static struct uart_msp432p4xx_dev_data_t uart_cc32xx_dev_data_0 = {
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+	.cb = NULL,
+#endif
+};
+
+static int uart_msp432p4xx_init(struct device *dev)
+{
+	const struct uart_device_config *config = DEV_CFG(dev);
+	eUSCI_UART_Config UartConfig;
+
+	/* Select P1.2 and P1.3 in UART mode */
+	MAP_GPIO_setAsPeripheralModuleFunctionInputPin(GPIO_PORT_P1,
+		(GPIO_PIN2 | GPIO_PIN3), GPIO_PRIMARY_MODULE_FUNCTION);
+
+	/* 115200 Buad rate settings calculated for 48MHz */
```

Comment:
We should be getting baud rate from the device tree.

This patch should have a YAML associated with it.